week1/homework/src/server.js

```diff
@@ -10,6 +10,32 @@ function createServer(port) {
 
   const server = http.createServer((request, response) => {
     // TODO: Write your homework code here
+    if (request.url === '/state') {
+      response.writeHead(200, { 'Content-Type': 'application/json' });
+      const main = { 'state': state };
+      response.end(JSON.stringify(main));
+    }
+    else if (request.url === '/add') {
+      response.writeHead(200, { 'Content-Type': 'application/json' });
```

Comment:
You are writing a head and writing to the response in every case of the if-statement: this is duplicate code; you should use a function for this.